pkg/sdkserver/sdkserver.go

```diff
@@ -985,41 +999,238 @@ func (s *SDKServer) GetList(ctx context.Context, in *alpha.GetListRequest) (*alp
+	defer s.gsUpdateMutex.Unlock()
+
+	gs, err := s.gameServer()
+	if err != nil {
+		return nil, err
+	}
+
+	if list, ok := gs.Status.Lists[in.Name]; ok {
+		batchList := s.gsListUpdates[in.Name]
+		// Verify room to add another value
+		var capacity int64
+		if batchList.capacitySet != nil {
+			capacity = *batchList.capacitySet
+		} else {
+			capacity = int64(len(list.Values) + len(batchList.listAppend) - len(batchList.listDelete))
+		}
+		if list.Capacity <= capacity {
+			return nil, errors.Errorf("OUT_OF_RANGE. No available capacity. Current Capacity: %d, List Size: %d", list.Capacity, len(list.Values))
+		}
+		// Verify value does not already exist in the list
+		// TODO: This does not check batched but not yet applied append / remove values. Should we do this?
+		// (Easy to check not yet applied values, hard to check removed and re-added values.) I'm
+		// thinking this would be better / easier to do as part of the batch apply update to list, and
+		// not verify here.
+		for _, val := range list.Values {
+			if in.Value == val {
+				return nil, errors.Errorf("ALREADY_EXISTS. Value: %s already in List: %s", in.Value, in.Name)
+			}
+		}
+		batchList.listAppend = append(batchList.listAppend, in.Value)
+		s.gsListUpdates[in.Name] = batchList
+		// Queue up the Update for later batch processing by updateLists.
+		s.workerqueue.Enqueue(cache.ExplicitKey(updateLists))
+		return &alpha.List{}, nil
+	}
+	return nil, errors.Errorf("NOT_FOUND. %s List not found", in.Name)
 }
 
-// RemoveListValue returns the updated List.
+// RemoveListValue collapses all remove a value from a List requests into a single UpdateList request.
+// Returns NOT_FOUND if the List does not exist.
+// Returns NOT_FOUND if the value is not in the List.
 // [Stage:Alpha]
 // [FeatureFlag:CountsAndLists]
 func (s *SDKServer) RemoveListValue(ctx context.Context, in *alpha.RemoveListValueRequest) (*alpha.List, error) {
 	if !runtime.FeatureEnabled(runtime.FeatureCountsAndLists) {
 		return nil, errors.Errorf("%s not enabled", runtime.FeatureCountsAndLists)
 	}
-	// TODO(#2716): Implement Me
-	return nil, errors.Errorf("Unimplemented -- RemoveListValue coming soon")
+
+	s.logger.WithField("name", in.Name).Debug("Remove List Value")
+
+	s.gsUpdateMutex.Lock()
+	defer s.gsUpdateMutex.Unlock()
+
+	gs, err := s.gameServer()
+	if err != nil {
+		return nil, err
+	}
+
+	if list, ok := gs.Status.Lists[in.Name]; ok {
+		// Verify value exists in the list
+		for _, val := range list.Values {
+			if in.Value != val {
+				continue
+			}
+			// Add value to remove to gsListUpdates map.
+			batchList := s.gsListUpdates[in.Name]
+			batchList.listDelete = append(batchList.listDelete, in.Value)
+			s.gsListUpdates[in.Name] = batchList
+			// Queue up the Update for later batch processing by updateLists.
+			s.workerqueue.Enqueue(cache.ExplicitKey(updateLists))
+			return &alpha.List{}, nil
+		}
+		return nil, errors.Errorf("NOT_FOUND. Value: %s not found in List: %s", in.Value, in.Name)
```

Comment:
From the [design](https://github.com/googleforgames/agones/issues/2716#:~:text=Returns%20false%20if%20the%20value%20is%20not%20found%20in%20the%20list%20(to%20the%20latest%20knowledge%20of%20the%20SDK)%2C), similar to append

> Returns false if the value is not found in the list (to the latest knowledge of the SDK),

So potentially a tweak on return types.

Reply:
It returns false from the client SDK.